src/util/imagefiledata.cpp

```diff
@@ -0,0 +1,26 @@
+#include "util/imagefiledata.h"
+
+#include <QByteArray>
+#include <QDebug>
+#include <QFile>
+#include <QFileInfo>
+#include <QImage>
+
+ImageFileData::ImageFileData(const QString& coverArtFilePath)
+        : QImage(coverArtFilePath) {
+    m_coverArtExtension = QFileInfo(coverArtFilePath).suffix();
+    QFile coverArtFile = QFile(coverArtFilePath);
+    coverArtFile.open(QIODevice::ReadOnly);
+    m_coverArtBytes = coverArtFile.readAll();
+    coverArtFile.close();
+}
+
+// Most of the cover art extensions are 'jpg' at cover art archive.
+ImageFileData::ImageFileData(const QByteArray& coverArtBytes)
+        : QImage(QImage::fromData(coverArtBytes)), m_coverArtBytes(coverArtBytes) {
+}
+
+// Could this be replaced with the saveCoverArt() function in coverartworker?
```

Comment:
you can just 
``
QFile file(basePath + format extension );
file.open(QIODevice::WriteOnly);
file.write(m_coverArtBytes);
file.close();
``

Reply:
The threading can be done outside of this class.

Reply:
Thank you! Then we can replace it with the function in CoverArtWorker's anon namespace IMHO

Reply:
> The threading can be done outside of this class.

So does it mean we don't need to do any disk work in this class? only caching?

Reply:
I've moved the function saveCoverArt() in cover art worker namespace into here.

Reply:
> So does it mean we don't need to do any disk work in this class? only caching?

My idea was to have ImageFileData::saveFile() as simple QImage::save() replacement, without any atomic or threading logic.

Reply:
I guess now it is already replaced with the QImage::save(), right? Only the extension , format is missing. I guess...